aws-http-client-url-connection/src/main/java/software/amazon/awssdk/http/java/UrlConnectionHttpClientFactory.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.java;
+
+import static software.amazon.awssdk.http.SdkHttpConfigurationOption.CONNECTION_TIMEOUT;
+import static software.amazon.awssdk.http.SdkHttpConfigurationOption.SOCKET_TIMEOUT;
+
+import java.time.Duration;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.SdkHttpClientFactory;
+import software.amazon.awssdk.http.SdkHttpConfigurationOption;
+import software.amazon.awssdk.utils.AttributeMap;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * A factory for an instance of {@link SdkHttpClient} that uses JDKs build-in {@link java.net.URLConnection} HTTP implementation.
+ * The factory can be configured via the {@link #builder()}, once built it can be use to create a {@link SdkHttpClient} via
+ * the {@link #createHttpClient()} method.
+ *
+ * <pre class="brush: java">
+ * SdkHttpClient httpClient = UrlConnectionHttpClientFactory.builder()
+ * .socketTimeout(Duration.ofSeconds(10))
+ * .connectionTimeout(Duration.ofSeconds(1))
+ * .build()
+ * .createHttpClient();
+ * </pre>
+ */
+public final class UrlConnectionHttpClientFactory
```

Comment:
For Apache and Netty we are using the SdkHttpClientFactory suffix, should we use that here?